libs/ast_generic/AST_generic.ml

```diff
@@ -2405,3 +2395,48 @@ let is_metavar_name s = Common.( =~ ) s "^\\(\\$[A-Z_][A-Z_0-9]*\\)$"
 (* coupling: Metavariable.is_metavar_ellipsis *)
 let is_metavar_ellipsis s =
   Common.( =~ ) s "^\\(\\$\\.\\.\\.[A-Z_][A-Z_0-9]*\\)$"
+
+(*****************************************************************************)
+(* Custom visitors *)
+(*****************************************************************************)
+
+(* Most clients should use this instead of the default `iter`. In many cases,
+ * it's not desirable to recurse into id_info since it contains resolved names
+ * and svalues which often contain nodes that are already present elsewhere in
+ * the AST. This matches the default behavior of the old mk_visitor. *)
+class virtual ['self] iter_no_id_info =
+  object (_self : 'self)
+    inherit ['self] iter
+    method! visit_id_info _env _info = ()
+  end
+
+(* This is based on the old behavior of Map_AST.mk_visitor. If you are adding a
+ * new callsite, you should consider whether the nonstandard behavior here makes
+ * sense for your use case. Think about using the autogenerated AST_generic.map
+ * directly. *)
+class virtual ['self] map_custom =
```

Comment:
I'd call this `Map_AST_default` instead or something similar

Reply:
Hmm, maybe I'll call it `map_legacy`? I'm not really sure exactly what to call it. Its defining characteristic is that it exists to mimic the previous behavior.